lxd/api_1.0.go

```diff
@@ -526,6 +526,17 @@ func doApi10Update(d *Daemon, r *http.Request, req api.ServerPut, patch bool) re
 		}
 	}
 
+	// Validate listener addresses and assign the default port if none given.
+	newCoreAddress, ok := nodeValues["core.https_address"]
+	if ok && newCoreAddress.(string) != "" {
+		nodeValues["core.https_address"] = util.CanonicalNetworkAddress(newCoreAddress.(string))
+	}
+
+	newClusterAddress, ok := nodeValues["cluster.https_address"]
+	if ok && newCoreAddress.(string) != "" {
+		nodeValues["cluster.https_address"] = util.CanonicalNetworkAddress(newClusterAddress.(string))
+	}
+
```

Comment:
We usually try to avoid having user provided values get rewritten, this can end up looking quite confusing.

Instead we should make sure we use the correct default port at the time we consume the configuration.

Reply:
I think I'll just drop this commit from this PR and move that over to its own PR then, because I suspect the diff will be a bit bigger.

Reply:
Turns out it was just 1 file after all so I'll keep it in :)

I did notice a sneaky little bug where everything would work until you restarted the daemon, after which `cluster.https_address` would again only be checked if `core.https_address` was non-empty, so that's taken care of now as well.